### PR TITLE
Introduce simple CI for pushes and PRs

### PR DESCRIPTION
**Pull request by adrianreber**
This runs 'cargo build', 'cargo clippy' and 'cargo fmt' to follow common Rust coding guidelines.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -28,7 +28,7 @@ fn set_realtime() {
 fn find_config() -> Result<String, Error> {
     let config_file_name = "buzzd.json";
     let home_direcory = env::var("HOME");
-    let global_config = format!("/etc/{}", config_file_name);
```
Comment (axkg) [inline src/main.rs:31]: I *did* run clippy and fmt, but neither complained about this. Was this a tool finding or did your Rustacean instincts trigger?

Comment (adrianreber) [reply src/main.rs:31]: One of changes, the line break, was triggered by cargo fmt. The inlining of the variable was done by clippy.

Maybe my rust version is newer.

File: src/main.rs [hunk 2]
```diff
@@ -112,9 +112,7 @@ fn play_pattern(pin: &mut OutputPin, config: &json::Value, pattern: &str, repeat
 }
 
 fn setup_mqtt_client(config: &json::Value) -> mqtt::Client {
-    let mqtt_broker = config["mqtt"]["broker"]
```
Comment (axkg) [inline src/main.rs:115]: Also strange, I did run fmt a couple of times and I could've sworn that it actually decided to spread that statement over multiple lines. Probably I removed something and now it is acceptable on a single line again...

**Review by axkg — APPROVED** [review]
Nice, thank you!

File: src/main.rs (hunk 1 above)
Comment (axkg) [reply src/main.rs:31]: OK, let's see what happens with the next CI run then 🤔  I did run fmt/clippy on ubuntu-22.04 (as does the workflow you added).

Comment (adrianreber) [reply src/main.rs:31]: There is a green ✅ next to my commit. Seems it was happy.

**Pull request merged**